doc/guides/zephyr_cmake_package.rst

```diff
@@ -382,6 +382,48 @@ Such a CMakeLists.txt could look as:
 
    project(my_first_app)
 
+.. _cmake_extension_package:
+
+Zephyr Extension CMake package
+******************************
+
+The Zephyr Extension CMake package provides a possibility for a Zephyr based system to control
+Zephyr build settings in a generic way.
+The Zephyr Extension CMake package will be loaded in the Zephyr boilerplate code after initial
+properties and ``ZEPHYR_BASE`` has been defined, but before CMake code execution.
+This allows the Zephyr Extension CMake package to setup or extend properties such as: ``DTS_ROOT``,
+``BOARD_ROOT``, Fixed overlays, and any other property that can be controlled.
+
+To provide a Zephyr Extension CMake package, create ``ZephyrExtensionConfig.cmake`` and place it in
+a Zephyr workspace top-level folder as:
+
+.. code-block:: none
+
+   <projects>/zephyr-workspace
+   ├── zephyr
+   ├── ...
+   └── zephyr_extension_project (can be named anything)
+        └── share/zephyrextension-package/cmake/ZephyrExtensionConfig.cmake
```

Comment:
Can it be just `share/zephyrextension-package/cmake/config.cmake` ?

Reply:
Unfortunately not.
From: https://cmake.org/cmake/help/latest/command/find_package.html#full-signature-and-config-mode
> By default the command searches for a package with the name `<PackageName>`. If the NAMES option is given the names following it are used instead of `<PackageName>`. The command searches for a file called `<PackageName>Config.cmake` or `<lower-case-package-name>-config.cmake` for each name specified.

Reply:
Thanks for explanation, I like  <lower-case-package-name>-config.cmake

Reply:
> 
> 
> Thanks for explanation, I like lower-case-package-name-config.cmake

In general, me too. But mostly I have experienced CamelCasing for CMake packages, which were why that was used here: https://github.com/zephyrproject-rtos/zephyr/tree/master/share/zephyr-package/cmake
and thus the reason for going with that in the example, although you may use `lower-case-package-name-config.cmake` in your local repo 😉